client/src/keyring.rs

```diff
@@ -25,8 +25,21 @@ pub enum Keyring {
 }
 
 impl Keyring {
+    /// Create a new instance of the Keyring that automatically removes the
+    /// broken items from the file backend keyring.
+    ///
+    /// This method will probably be removed in future versions if the
+    /// mis-behaviour is tracked and fixed.
+    pub async fn with_auto_delete_broken_items() -> Result<Self> {
```

Comment:
@zecakeh this should make things easier for us, but the name is very bad...

Reply:
Ah nice !